contrib/native/client/CMakeLists.txt

```diff
@@ -25,6 +25,13 @@ cmake_policy(SET CMP0043 NEW)
 cmake_policy(SET CMP0048 NEW)
 enable_testing()
 
+#
+# required version for dependencies
+#
+set (BOOST_MINIMUM_VERSION 1.54.0)
+set (PROTOBUF_MINIMUM_VERSION 3.6.1)
```

Comment:
The current version uses protobuf 3.11 version (see readme.linux).

Reply:
`readme.win.txt` mentions 3.6.1, which is why I chose that version. I don't think it matters that much, I just wanted to make sure `ByteSizeLong()` method was present.

Reply:
It may cause issues for other places since protobuf files were regenerated using 3.11 version. Looks like updating `readme.win.txt` instruction was missed and should be fixed to be up-to-date.

Reply:
I don't think this is quite true on C++ code, files can be regenerated with whatever version is present on the system (it's even explained on how to do it). Apart from bugfixes, I don't think there are incompatibilities introduced between protobuf minor versions.

Currently I cannot validate windows build. If someone can do it quickly, I can sure change it, otherwise I'd prefer to get it addressed separately.

Reply:
Here is an example of the issue when another version was used for generating protobuf files: https://github.com/apache/drill/pull/2000#issuecomment-592024872. Your change would significantly simplify finding the root cause of such issues.

By the way, here are changes in protobuf files for updating to 3.11.1 version: https://github.com/apache/drill/commit/7e6fc81ef414cdeca8120c20e12fcce21893f7bf#diff-b5a38f368686e0d3202ec29cd8276d6f

So it may be possible that these newer changes use some new methods introduced after 3.6.

Regarding validating windows build, I agree that it can be done separately.

Reply:
Looks like regenerating with the correct version present on the user machine would be enough to fix the issue, but Drill has protobuf generated classes merged into code base instead of having those generated on the fly (not sure if people would be willing to revisit this? protoc compilers are now available in maven central, so you don't need to have installed on the system to build the java code).

Reply:
protoc compiler is required to be installed when regenerating c++ protobuf files, not java classes, therefore I pointed to that PRs in my previous comment.

But this change doesn't make things worse, so it may be merged as it is.

Reply:
@vvysotskyi @laurentgo 
Can we merge this PR?

Reply:
Yes, it can be merged, but after resolving conflicts.